### ISSUE TITLE
Problem with Cost Item Resource Value

### ISSUE THREAD

**Issue by maxfb87** [IfcOpenShell/IfcOpenShell#4122]
I think there is a bug with hours/day conversion in Cost Item Resource Value.

Open the IFC file attached.
There is a Wall with a GrossVolume 3mc. I assigned a labour resource (fixed cost 35€) that has a productivity rate 1 GrossVolume / 7 hours.
So the man hours in this case is correctly 21h to construct the wall.
After that, i click on the button "calculate cost item resource value" and correctly the cost item value is 735€ (35 x 21 = 735).

If i change the productivity into 1 GrossVolume / 8 hours, the man hours change into 24h, but when i click on "calculate cost item resource value", the cost item value is 35€, that is obviously wrong ...
[prova_5d.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/13687757/prova_5d.txt)
 it seems that it converts 24h into 1 day and it multiply 1 x 35 ...

Am i missing something or is it a bug?
@myoualid

**Comment by myoualid** [comment]
You are right, there was a bug now fixed. Tough there is still an assumption  to get rid of in this code: replace an 8 hour day with the resource's calendar, so let's leave this open for now

**Comment by BIMvoice** [comment]
@myoualid addressed the remaining calendar assumption you flagged: the hardcoded 8-hour day is now replaced with a lookup of the resource's actual IfcWorkCalendar (own assignment, parent resource, or assigned task), falling back to 8 hours only when no calendar data exists. Proposed in #8616.

Generated with the assistance of an AI coding tool.
